pkg/etcdcli/etcdcli.go

```diff
@@ -274,6 +275,19 @@ func (g *etcdClientGetter) GetMember(name string) (*etcdserverpb.Member, error)
 	return nil, apierrors.NewNotFound(schema.GroupResource{Group: "etcd.operator.openshift.io", Resource: "etcdmembers"}, name)
 }
 
+// If the member's name is not set, extract ip/hostname from peerURL. Useful with unstarted members.
+func GetMemberNameOrHost(member *etcdserverpb.Member) string {
+	if len(member.Name) == 0 {
+		u, err := url.Parse(member.PeerURLs[0])
+		if err != nil {
+			klog.Errorf("unstarted member has invalid peerURL: %#v", err)
+			return "NAME-PENDING-BAD-PEER-URL"
```

Comment:
If we have a getter that can have an err it should be returned. I mean the whole idea of an error in this context is to exit and retry. I know inlining is nice but you are not covering your failure.

member, err  := GetMemberNameOrHost()

Reply:
Since we are not the original composers of the member list (it came out of a client call), I think exiting is not required. We can display error and then on the next iteration, we get a new member list and hopefully the problem is fixed. Exiting the operator doesn't fix a problem with etcd (if it is indeed a problem with etcd's member list).

Reply:
making the operator resync on a parse error is probably heavy-handed given the context of the error. I guess what bothers me is I can't think of a case where this parse error would happen in a sane way. This looks better in general.